matlab/HyEQsolver_new_slow.m

```diff
@@ -0,0 +1,434 @@
+%         ylabel('x1')
+%
+%         figure(2) % velocity
+%         clf
+%         subplot(2,1,1),plotflows(t,j,x(:,2))
+%         grid on
+%         ylabel('x2')
+%
+%         subplot(2,1,2),plotjumps(t,j,x(:,2))
+%         grid on
+%         ylabel('x2')
+%
+%         % plot hybrid arc
+%         
+%         figure(3)
+%         plotHybridArc(t,j,x)
+%         xlabel('j')
+%         ylabel('t')
+%         zlabel('x1')
+%
+%         % plot solution using plotHarc and plotHarcColor
+%
+%         figure(4) % position
+%         clf
+%         subplot(2,1,1), plotHarc(t,j,x(:,1));
+%         grid on
+%         ylabel('x_1 position')
+%         subplot(2,1,2), plotHarc(t,j,x(:,2));
+%         grid on
+%         ylabel('x_2 velocity')
+%
+%
+%         % plot a phase plane
+%         figure(5) % position
+%         clf
+%         plotHarcColor(x(:,1),j,x(:,2),t);
+%         xlabel('x_1')
+%         ylabel('x_2')
+%         grid on
+%
+%--------------------------------------------------------------------------
+% Matlab M-file Project: HyEQ Toolbox @ Hybrid Systems Laboratory (HSL),
+% https://hybrid.soe.ucsc.edu/software
+% http://hybridsimulator.wordpress.com/
+% Filename: HyEQsolver.m
+%   Copyright @ Hybrid Systems Laboratory (HSL),
+%   Revision: 3.0.0 Date: 08/14/2021
+%--------------------------------------------------------------------------
+% See also HybridSystem, HybridPlotBuilder, 
+% <a href="matlab:hybrid.internal.openHelp('HybridSystem_demo')">Demo: How to Implement and Solve a Hybrid System</a>.
+% 
+% Added prior to HyEQ Toolbox version 2.04. Major revision in v3.0.
+
+
+if ~exist('rule','var')
+    rule = 1;
+end
+
+if ~exist('options','var')
+    options = odeset();
+end
+if exist('E','var') && ~exist('solver','var')
+    solver = 'ode15s';
+end
+if ~exist('solver','var') || isempty(solver)
+    solver = 'ode45';
+end
+if ~exist('E','var')
+    E = [];
+end
+if ~exist('progress', 'var')
+    progress = hybrid.PopupProgressUpdater();
+elseif strcmp(progress, 'silent')
+    progress = hybrid.SilentProgressUpdater();
+end
+% mass matrix (if existent)
+isDAE = false;
+if ~isempty(E)
+    isDAE = true;
+    switch isa(E,'function_handle')
+        case true % Function E(x)
+            M = E;
+            options = odeset(options,'Mass',M,'Stats','off',...
+                'MassSingular','maybe','MStateDependence','strong',...
+                'InitialSlope',f_hdae(x0,TSPAN(1))); 
+        case false % Constant double matrix
+            M = double(E);
+            options = odeset(options,'Mass',M,'Stats','off',...
+                'MassSingular','maybe','MStateDependence','none');
+    end
+end
+
+options = odeset(options, 'OutputFcn', @ODE_callback);
+
+odeX = str2func(solver);
+f = wrap_with_3args(f);
+g = wrap_with_3args(g);
+C = wrap_with_3args(C);
+D = wrap_with_3args(D);
+
+% simulation horizon
+tstart = TSPAN(1);
+tfinal = TSPAN(end);
+
+% Preallocate memory to speed up simulations
+prealloc_size = 1e6;
+index_last = 1;
```

Comment:
A more descriptive name for `index_last` or a comment describing it would be helpful. What is it the last _of_?